[WIP] Mission view

EuniceNam opened the pull request:
## 작업 내용

- [ ] UI 미션 화면 만들기
  - [x] 진행도 뷰 만들기
  - [x] 카드 뷰 만들기
  - [x] 회의종료 버튼 추가하기
  - [ ] UI 공용 박스, 선, 진행도 두께 적용
- [ ] 파이어베이스 연결하기


## 고민한 내용
- 카드 뷰를 따로 만드는 게 맞는지 모르겠어요
- 무새 이미지 크기 조정이 필요한 것 같아요

## 리뷰 받고 싶은 내용 ( 질문할 내용 )
- 이미지 크기 조정(확대, 크롭) 어떻게 할 수 있나요?
- 카드 뷰를 따로 만드는 게 맞는지 모르겠어요 -> 해결
  - 무새 이미지 크기 조정이 어려운 느낌
  - 캔버스 로딩은 좀 느린 것 같음

![image](https://user-images.githubusercontent.com/18394923/173193152-e77b8b70-386a-42a2-97e4-9935044dbc4c.png)

EuniceNam committed:
#11 add MissionProgressCircleView

EuniceNam committed:
fix typo: view name

EuniceNam committed:
#12 #11 add MissionView

cozytk commented on MeetingNMusae/MeetingNMusae/Views/MissionCardView.swift:
```diff
@@ -0,0 +1,75 @@
+            Text(roleName)
+
+            // will be changed to a custom line - based on 역할 선택 (상세) view
+            Divider()
```
```swift
//
//  Line.swift
//  MeetingNMusae
//
//  Created by taekkim on 2022/06/11.
//

import SwiftUI

struct Line: Shape { func path(in rect: CGRect) -> Path {
        var path = Path()
        path.move(to: CGPoint(x: 0, y: 0))
        path.addLine(to: CGPoint(x: rect.width, y: 0))
        return path
    }
}
```
Views에 Line 이라는 파일에 Line struct 만들어놨습니다


```suggestion
            Line()
                .stroke(style: StrokeStyle(lineWidth: 3, dash: [10]))
                .frame(height: 1)
```

요롷게 쓰시면 될 거 같아여!!!

EuniceNam replied:
감사합니다. 수정했습니다!
0d4acf84dbeeb11e3bbaf99619a27f9c1fb6b62d

cozytk commented:
카드 뷰를 따로 만드는 게 맞는지 모르겠어요
->
https://github.com/DeveloperAcademy-POSTECH/MC2-Team15-MeetingNMusae/pull/18 
이 뷰랑 같이 쓸거면 따로 만들면 좋을 것 같은데 아니라면 `@ViewBuilder` 로 만들면 좋을 것 같습니다!!
https://dev200ok.blogspot.com/2022/05/swiftui.html

EuniceNam committed:
#12 refactor: struct subview to viewbuilder

plus adding UIScreen extension following Cozy's UIScreen+Extensions file

According to code review: https://github.com/DeveloperAcademy-POSTECH/MC2-Team15-MeetingNMusae/pull/19#issuecomment-1152866389

EuniceNam committed:
#12 feat: dashed line

according to
https://github.com/DeveloperAcademy-POSTECH/MC2-Team15-MeetingNMusae/pull/19#discussion_r894983541

EuniceNam commented:
좋은 자료 감사합니다. 
여기서만 쓸 것 같아 뷰빌더로 수정했는데, 뷰빌더 안에는 프로퍼티 래퍼를 쓸 수 없는 것 같아서 이후 서버 연결에서 뷰빌더에 넘길 데이터가 많아지면 바꿀 수도 있을 것 같아요. 
e086543cfe79b8c6d180a66e54c07c445de66a2b

EuniceNam committed:
feat: an end-meeting button

EuniceNam committed:
Refactor and add temporal CharacterBox

CharacterBox from 1ccab00ff424c2b053ab4175159d8d22d9637466

Plus refactor other branches' code to a separated file

EuniceNam committed:
#12 fix: mission box shadow color

tea-hkim commented:
뷰 빌더 만드는 거 말고 밑에 Extract View로 따로 빼도 될 것 같아요

EuniceNam commented:
현재 메인에서 다시 브랜치 팔게요
